main.c

```diff
@@ -235,20 +236,82 @@ static bool path_exists(const char *path) {
 	return path && access(path, R_OK) != -1;
 }
 
-static const char *get_output_dir(void) {
-	static const char *output_dirs[] = {
-		"GRIM_DEFAULT_DIR",
-		"XDG_PICTURES_DIR",
-	};
+char *get_xdg_pictures_dir(void) {
+	char *home_dir = getenv("HOME");
+	if (home_dir == NULL) {
+		return NULL;
+	}
+
+	char *config_file;
+	char *config_home = getenv("XDG_CONFIG_HOME");
+	if (config_home == NULL || config_home[0] == 0) {
+		config_file = (char *)malloc(strlen(home_dir) + strlen("/.config/user-dirs.dirs") + 1);
+		if (config_file == NULL) {
+			return NULL;
+		}
+		strcpy(config_file, home_dir);
+		strcat(config_file, "/.config/user-dirs.dirs");
```

Comment:
Please avoid functions like `strcpy` and `strcat`. Prefer the bounds-checking versions, or alternatively `snprintf`.